Rpackage/R/mmit.pruning.R

```diff
@@ -0,0 +1,83 @@
+#' The Pruned Max Margin Interval Tree
+#'
+#' Pruning the regression tree for censored data.
+#'
+#' @param tree The fitted tree using \code{"mmit()"} function
+#' 
+#' @return The learned regression tree as an object of class party.
+#' 
+#' @author Toby Dylan Hocking, Alexandre Drouin, Torsten Hothorn, Parismita Das
+#' 
+#' @examples
+#' library(mmit)
+#' target.mat <- rbind(
+#'   c(0,1), c(0,1), c(0,1),
+#'   c(2,3), c(2,3), c(2,3))
+#' 
+#' feature.mat <- rbind(
+#'   c(1,0,0), c(1,1,0), c(1,2,0),
+#'   c(1,3,0), c(1,4,0), c(1,5,0))
+#' 
+#' colnames(feature.mat) <- c("a", "b", "c")
+#' feature.mat <- data.frame(feature.mat)
+#' 
+#' 
+#' tree <- mmit(target.mat, feature.mat)
+#' 
+#' pruned_tree <- mmit.pruning(tree)
+#' 
+#' @export
+
+
+
+#### needs work...eturn type
+
+
+
+mmit.pruning <- structure(function(tree){
+  ### T1 is the tree after initital pruning of Tmax
+  T1 <- init_pruning(tree)
+  alpha_trees <- NULL
+  
+  sequential_prune <- function(tree){
+    ### find weakest links
+    wlink <- weakest_link(tree)
+    
+    ### min_gt is same for all weakest links
+    min_gt <- wlink[1, 1]
+    
+    ### to prune the weakest links from tail of tree we apply sorting
+    ### sort solution 
+    sorted_wlink <- order(wlink[, 2])
+    if(length(sorted_wlink)>1){
+      wlink <- wlink[sorted_wlink,]
+    }
+
+    for(n in nrow(wlink):1){
+      tree <- nodeprune(tree, ids = wlink[n, 2])
```

Comment:
Why don't you simply call nodeprune once with all ids to prune? The function can handle a vector of ids. You would avoid all potential bugs due to IDs changing after pruning.

Reply:
ok